cmd/askd/routes/routes.go

```diff
@@ -100,71 +85,74 @@ func API() http.Handler {
 	return a
 }
 
-// oldRoutes manages the handling of the API endpoints for the old style ported
-// from Ask.
-//
-// FIXME: remove on next API release
-func oldRoutes(a *app.App) {
-	// forms
-	a.Handle("POST", "/api/form", handlers.Form.Upsert)
-	a.Handle("PUT", "/api/form", handlers.Form.Upsert)
-	a.Handle("PUT", "/api/form/:id/status/:status", handlers.Form.UpdateStatus)
-	a.Handle("GET", "/api/forms", handlers.Form.List)
-	a.Handle("GET", "/api/form/:id", handlers.Form.Retrieve)
-	a.Handle("DELETE", "/api/form/:id", handlers.Form.Delete)
+func routes(a *app.App) {
 
-	// form submissions
-	a.Handle("POST", "/api/form_submission/:id", handlers.FormSubmission.Create)
-	a.Handle("PUT", "/api/form_submission/:id/status/:status", handlers.FormSubmission.UpdateStatus)
-	a.Handle("GET", "/api/form_submissions/:form_id", handlers.FormSubmission.Search)
-	a.Handle("GET", "/api/form_submission/:id", handlers.FormSubmission.Retrieve)
-	a.Handle("PUT", "/api/form_submission/:id/:answer_id", handlers.FormSubmission.UpdateAnswer)
-	a.Handle("PUT", "/api/form_submission/:id/flag/:flag", handlers.FormSubmission.AddFlag)
-	a.Handle("DELETE", "/api/form_submission/:id/flag/:flag", handlers.FormSubmission.RemoveFlag)
-	a.Handle("DELETE", "/api/form_submission/:id", handlers.FormSubmission.Delete)
+	// Create a new app group which will be for internal functions that may have
+	// an optional layer of auth added to it.
+	internal := a.Group()
 
-	// form galleries
-	a.Handle("GET", "/api/form_gallery/:id", handlers.FormGallery.Retrieve)
-	a.Handle("GET", "/api/form_galleries/:form_id", handlers.FormGallery.RetrieveForForm)
-	a.Handle("GET", "/api/form_galleries/form/:form_id", handlers.FormGallery.RetrieveForForm)
-	a.Handle("PUT", "/api/form_gallery/:id/add/:submission_id/:answer_id", handlers.FormGallery.AddAnswer)
-	a.Handle("PUT", "/api/form_gallery/:id", handlers.FormGallery.Update)
-	a.Handle("DELETE", "/api/form_gallery/:id/remove/:submission_id/:answer_id", handlers.FormGallery.RemoveAnswer)
-}
+	// Now we will load in the public key from the config. If found, we'll add a
+	// middleware to all internal endpoints that will ensure that we validate the
+	// requests coming in.
+
+	publicKey, err := cfg.String(cfgAuthPublicKey)
+	if err != nil || publicKey == "" {
+		log.User("startup", "Init", "XENIA_%s is missing, internal authentication is disabled", cfgAuthPublicKey)
```

Comment:
I didn't change the XENIA_ yet because ask is already in production/staging and this would break that. When we're ready to change it, we could bundle it into it's own PR.